### PR TITLE
Fix Bug 1456082 - Storage should not have telemetry be a required argument

### PR DESCRIPTION
**Pull request by piatra**
Telemetry is no longer required and updated unit test that checks for that.

### CODE REVIEW

File: system-addon/lib/ActivityStreamStorage.jsm [hunk 1]
```diff
@@ -4,15 +4,15 @@ this.ActivityStreamStorage = class ActivityStreamStorage {
   /**
    * @param storeNames Array of strings used to create all the required stores
    */
-  constructor(options = {}) {
-    if (!options.storeNames || !options.telemetry) {
-      throw new Error(`storeNames and telemetry are required, called only with ${Object.keys(options)}`);
+  constructor({storeNames, telemetry}) {
```
Comment (k88hudson) [inline system-addon/lib/ActivityStreamStorage.jsm]: won't your request wrapper throw if telemetry is falsey?

Comment (piatra) [reply system-addon/lib/ActivityStreamStorage.jsm]: It is only used if non falsey.

https://github.com/mozilla/activity-stream/blob/106c699eed2f594dc280806d684ca2fdf3389f94/system-addon/lib/ActivityStreamStorage.jsm#L73

**Comment by as-pine-proxy** [comment]
[Treeherder: 2580b20f6567595fb276a90e34f14aed44b3d6de](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=2580b20f6567595fb276a90e34f14aed44b3d6de)

**Comment by as-pine-proxy** [comment]
[Treeherder: 9b663ca485dce7b8b8759b43537bd85fdc9b31de](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=9b663ca485dce7b8b8759b43537bd85fdc9b31de)

**Pull request merged**